ocaml/xapi/helpers.ml

```diff
@@ -127,9 +127,12 @@ let get_management_ip_addr ~__context =
 let get_localhost_uuid () =
   Xapi_inventory.lookup Xapi_inventory._installation_uuid
 
-let get_localhost ~__context : API.ref_host  =
-  let uuid = get_localhost_uuid () in
-  Db.Host.get_by_uuid ~__context ~uuid
+let get_localhost ~__context : API.ref_host =
+  let localhost_ref = !Xapi_globs.localhost_ref in
+  match !Xapi_globs.localhost_ref = Ref.null with
```

Comment:
This could be an `if` statement

Reply:
The above ref is set to a value that is returned by this `get_localhost` function, while the function now inspects the ref. See https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/dbsync_slave.ml#L245. Checking whether this is safe makes my brain explode.

In general, before making a change like this, we need to be able to guarantee that all callers of this function are happy to receive a cached version of the ref. Otherwise things may break in unexpected places.

Reply:
Is it expected that localhost's Ref ever changes after it is set?
I agree that the potential infinite recursion is worrying, perhaps we should have a `get_localhost_uncached` and have the code that initially sets it call that instead to be simpler?

Reply:
will drop this commit and test on another branch